### ISSUE TITLE
Changing health check on TCP listener requires resource to be deleted and re-created

### ISSUE THREAD

**Issue by iancward** [terraform-providers/terraform-provider-aws#4150]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
```
Terraform v0.11.1
+ provider.aws v1.5.0
``

### Affected Resource(s)
- aws_lb_target_group

### Terraform Configuration Files
```hcl
resource "aws_lb_target_group" "http" {
  name                 = "${ local.resource_name }-80"
  port                 = 80
  protocol             = "TCP"
  vpc_id               = "${ module.network.vpc_id }"
  deregistration_delay = 300

  health_check {
    interval            = 10
    port                = 443
    protocol            = "HTTPS"
    path                = "/health"
    healthy_threshold   = 2
    unhealthy_threshold = 2
    matcher             = "200-399"
  }
}

resource "aws_lb_listener" "http" {
  load_balancer_arn = "${ aws_lb.main.arn }"
  port              = 80
  protocol          = "TCP"

  default_action = {
    target_group_arn = "${ aws_lb_target_group.http.arn }"
    type             = "forward"
  }
}
```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
AWS Currently does not let you change the health check type from one kind (e.g. TCP, HTTP, HTTPS) to another; the target group resource must be deleted and re-created

### Actual Behavior
Terraform received this error from the AWS API:
> Error modifying Target Group: InvalidConfigurationRequest: You cannot change the health check protocol for a target group with the TCP protocol

### Steps to Reproduce
1. Configure aws_lb_listener as TCP
2. Configure aws_lb_target_group with TCP health check
3. Run `terraform apply`
4. Change aws_lb_target_group created in step 2 above to have a HTTPS health check
5. Run `terraform apply`
6. Receive error.

### Important Factoids
N/A

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
https://github.com/terraform-providers/terraform-provider-aws/issues/2708#issuecomment-357355482

**Comment by catsby** [comment]
I believe this is a "known" limitation in the API. Terraform may now be better able to handle it though, with a `CustomizeDiff` method to make this a `ForceNew` operation if the protocol of the health check is TCP.

Unfortunately I don't know that anyone on the Terraform team will be able to pick this up anytime soon, so if anyone wants to take it up, that's where I'd start looking :D

**Comment by iancward** [comment]
It looks like there are already a few warnings for changing health check intervals on TCP listeners. I think we could just add another block for changing the listener type https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_lb_target_group.go#L609-L620.

**Comment by bflad** [comment]
The fix for this has been merged and will release in version 2.57.0 of the Terraform AWS Provider, later this week. Thanks to @iancward for the implementation. 👍
